src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -358,7 +365,16 @@ pub trait TimestampProvider {
         if when.can_advance_to_upper()
             && (isolation_level == &IsolationLevel::Serializable || timeline.is_none())
         {
-            candidate.join_assign(&largest_not_in_advance_of_upper);
+            let mut timestamp = largest_not_in_advance_of_upper;
+            // In practice advancing the timestamp into the future is confusing for users. So if
+            // there is an `oracle_read_ts` then we treat is as "the current time" bound the upper
+            // advancement to "the current time".
+            if isolation_level == &IsolationLevel::Serializable {
```

Comment:
[Our docs are saying](https://materialize.com/docs/get-started/isolation-level/#serializable) for the 4 standard levels (i.e., anything weaker or equal to SERIALIZABLE) that "In Materialize, you can request any of these isolation levels, but they all behave the same as the Serializable isolation level.". So, shouldn't this `if` check for `<= Serializable` instead of `== Serializable`?

Note that I've actually seen General Mills use `ReadUncommitted`...

Reply:
We never actually initialize the other isolation levels, they're all parsed as `Serializable`. That's a bit fragile so maybe we should update it, but that's how the rest of the code is structured.

Reply:
Oh, ok, I see!